Description: Remove Sphincs+ tests to build without PySPX
 PySPX is an optional runtime dependency to support post-quantum signing. It is
 not available as Debian package. This patch removes tests that require PySPX,
 to allow building securesystemslib. PySPX may still be installed via pip or the
 manylinux1 docker image, when using securesystemslib.
 https://github.com/sphincs/pyspx

Author: Lukas Puehringer <lukas.puehringer@nyu.edu>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/tests/test_keys.py
+++ b/tests/test_keys.py
@@ -33,27 +33,6 @@
 DATA = securesystemslib.formats.encode_canonical(DATA_STR).encode("utf-8")
 
 
-@unittest.skipIf(os.name == "nt", "PySPX n/a on Windows")
-class TestSphincsKeys(unittest.TestCase):
-    """Test create keys, sign and verify for sphincs keys."""
-
-    def test_sphincs_keys(self):
-        key = KEYS.generate_sphincs_key()
-        sig = KEYS.create_signature(key, b"data")
-        self.assertTrue(securesystemslib.formats.SIGNATURE_SCHEMA.matches(sig))
-
-        # Assert valid/invalid signature
-        self.assertTrue(KEYS.verify_signature(key, sig, b"data"))
-        self.assertFalse(KEYS.verify_signature(key, sig, b"not data"))
-
-        # Assert verificaiton failure for unsupported signing scheme
-        key["scheme"] = "invalid_scheme"
-        with self.assertRaises(
-            securesystemslib.exceptions.UnsupportedAlgorithmError
-        ):
-            KEYS.verify_signature(key, sig, b"data")
-
-
 class TestKeys(unittest.TestCase):  # pylint: disable=missing-class-docstring
     @classmethod
     def setUpClass(cls):
--- a/tests/test_signer.py
+++ b/tests/test_signer.py
@@ -280,8 +280,6 @@
             KEYS.generate_ed25519_key(),
             KEYS.generate_ecdsa_key(),
         ]
-        if os.name != "nt":
-            cls.keys.append(KEYS.generate_sphincs_key())
 
         cls.DATA = b"DATA"
 
